includes/functions.inc.php

```diff
@@ -26,39 +26,30 @@ function input_convert($str) {
 }
 
 
-function format_ago($time, $ago = false) {
+function format_ago($time) {
   $minute = 60;
   $hour   = $minute * 60;
   $day    = $hour   * 24;
 
   $when = $time;
 
-  if ($when >= 0)
-    $suffix = 'ago';
-  else {
-    $when = -$when;
-    $suffix = 'in the future';
-  }
-
   if ($when > $day) {
-    $when = round($when / $day);
-    $what = 'day';
+    $tmpday = floor($when / $day);
+    $tmphour = floor(($when / $hour) - (24*$tmpday));
+    $tmpminute = floor(($when / $minute) - (24*60*$tmpday) - ($tmphour * 60));
+    $tmpsec = floor($when - (24*60*60*$tmpday) - ($tmphour * 60 * 60) - ($tmpminute * 60));
+    return sprintf("%d day %d hour %d minute %d sec",$tmpday,$tmphour,$tmpminute,$tmpsec);
```

Comment:
This will now print things like `12 day` instead of `12 days` which looks bad. Can you please add the `s` if it's `!= 1` like it used to. You could do:
```php
return sprintf("%d hour%s %d minute%s %d second%s", $tmphour, ($tmphour != 1) ? 's' : '', $tmpminute, ($tmpminute != 1) ? 's' : '', $tmpsec, ($tmpsec != 1) ? 's' : '');
```